src/test/java/io/pravega/connectors/flink/FlinkPravegaOutputFormatTest.java

```diff
@@ -0,0 +1,191 @@
+                .withSerializationSchema(serializationSchema)
+                .withPravegaConfig(pravegaConfig)
+                .forStream(stream)
+                .build();
+    }
+
+    /**
+     * Testing the builder for right configurations.
+     * Should fail since we don't pass {@link SerializationSchema}
+     */
+    @Test(expected = NullPointerException.class)
+    public void testBuilderForFailure1() {
+        PravegaConfig pravegaConfig = mock(PravegaConfig.class);
+        FlinkPravegaOutputFormat.<String>builder()
+                .withEventRouter(eventRouter)
+                .withPravegaConfig(pravegaConfig)
+                .forStream(stream)
+                .build();
+    }
+
+    /**
+     * Testing the builder for right configurations.
+     * Should fail since we don't pass {@link PravegaEventRouter}
+     */
+    @Test(expected = NullPointerException.class)
+    public void testBuilderForFailure2() {
+        FlinkPravegaOutputFormat.<String>builder()
+                .withSerializationSchema(serializationSchema)
+                .withPravegaConfig(pravegaConfig)
+                .forStream(stream)
+                .build();
+    }
+
+    /**
+     * Testing the builder for right configurations.
+     * Should fail since we don't pass {@link Stream}
+     */
+    @Test(expected = IllegalStateException.class)
+    public void testBuilderForFailure3() {
+        FlinkPravegaOutputFormat.<String>builder()
+                .withEventRouter(eventRouter)
+                .withSerializationSchema(serializationSchema)
+                .withPravegaConfig(pravegaConfig)
+                .build();
+    }
+
+    /**
+     * Test case to validate the lifecycle methods of {@link OutputFormat}
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testLifecycleMethods() throws Exception  {
+        ClientConfig clientConfig = ClientConfig.builder().build();
+        when(pravegaConfig.getClientConfig()).thenReturn(clientConfig);
+        when(pravegaConfig.resolve(anyString())).thenReturn(stream);
+
+        EventStreamWriter<String> pravegaWriter = mockEventStreamWriter();
+        ClientFactory clientFactory = mockClientFactory(pravegaWriter);
+        FlinkPravegaOutputFormat<String> spyFlinkPravegaOutputFormat = spyFlinkPravegaOutputFormat(clientFactory);
+
+        // test open
+        spyFlinkPravegaOutputFormat.open(0, 1);
+        verify(spyFlinkPravegaOutputFormat).open(0, 1);
+
+        CompletableFuture<Void> writeFuture = new CompletableFuture<>();
+        when(pravegaWriter.writeEvent(anyString(), anyObject())).thenReturn(writeFuture);
+
+        ExecutorService executorService = spy(new DirectExecutorService());
+        Mockito.doReturn(executorService).when(spyFlinkPravegaOutputFormat).createExecutorService();
+
+        // test writeRecord
+        spyFlinkPravegaOutputFormat.writeRecord("test");
+        verify(spyFlinkPravegaOutputFormat).writeRecord("test");
```

Comment:
Not clear why you'd verify that `writeRecord` was called since the test itself calls it.  I imagine that the tests for `writeRecord` would mostly focus on the `pendingWritesCount`, `checkWriteError`, etc.

Reply:
(done)